ci/cloudfront-orchestration/waf/template.yaml

```diff
@@ -0,0 +1,184 @@
+        - Name: !Sub "${Environment}-oic-common-rule-set"
+          OverrideAction:
+            None: {}
+          Priority: 2
+          Statement:
+            ManagedRuleGroupStatement:
+              Name: "AWSManagedRulesCommonRuleSet"
+              VendorName: "AWS"
+              RuleActionOverrides:
+                - Name: "GenericRFI_QUERYARGUMENTS"
+                  ActionToUse:
+                    Count: {}
+                - Name: "GenericRFI_BODY"
+                  ActionToUse:
+                    Count: {}
+                - Name: "NoUserAgent_HEADER"
+                  ActionToUse:
+                    Count: {}
+                - Name: "EC2MetaDataSSRF_QUERYARGUMENTS"
+                  ActionToUse:
+                    Count: {}
+                - Name: "SizeRestrictions_QUERYSTRING"
+                  ActionToUse:
+                    Count: {}
+                - !If
+                  - IsProduction
+                  - Name: "EC2MetaDataSSRF_BODY"
+                    ActionToUse:
+                      Count: {}
+                  - !Ref AWS::NoValue
+          VisibilityConfig:
+            CloudWatchMetricsEnabled: true
+            MetricName: !Sub "${Environment}OidcWafCommonRuleSet"
+            SampledRequestsEnabled: true
+        - Name: !Sub "${Environment}-oidc-bad-rule-set"
+          OverrideAction:
+            None: {}
+          Priority: 3
+          Statement:
+            ManagedRuleGroupStatement:
+              Name: "AWSManagedRulesKnownBadInputsRuleSet"
+              VendorName: "AWS"
+          VisibilityConfig:
+            CloudWatchMetricsEnabled: true
+            MetricName: !Sub "${Environment}OidcWafBadRuleSet"
+            SampledRequestsEnabled: true
+        - !If
+          - IsProduction
+          - Name: "production-smoke-test-client-exception"
+            Action:
+              Block: {}
+            Priority: 4
+            Statement:
+              AndStatement:
+                Statements:
+                  - LabelMatchStatement:
+                      Key: "awswaf:managed:aws:core-rule-set:EC2MetaDataSSRF_QueryArguments"
+                      Scope: "LABEL"
+                  - NotStatement:
+                      Statement:
+                        ByteMatchStatement:
+                          FieldToMatch:
+                            SingleQueryArgument:
+                              Name: "client_id"
+                          PositionalConstraint: "EXACTLY"
+                          SearchString: !Ref SmokeTestClientId
+                          TextTransformations:
+                            - Priority: 0
+                              Type: "NONE"
+            VisibilityConfig:
+              CloudWatchMetricsEnabled: true
+              MetricName: "ProductionSmokeTestClientExceptionRule"
+              SampledRequestsEnabled: true
+          - !Ref AWS::NoValue
+        - Name: !Sub "${Environment}-default-query-param-limit"
```

Comment:
I don't know if you're bothered, but in the terraform this has underscores instead of dashes & doesn't have an environment (but yours looks better!)

Reply:
Yeah, I went for consistency with the others rather than keeping it the same. These are going to end up in a different region, so metrics will change anyway